src/services/CacheService.ts

```diff
@@ -758,1318 +745,116 @@ export default class UtilsService {
-                votingMachineProposalInfo.currentBoostedVotePeriodLimit;
-              networkCache.proposals[proposal.id].daoBountyRemain = bnum(
-                votingMachineProposalInfo.daoBountyRemain
-              );
-              networkCache.proposals[proposal.id].daoBounty = bnum(
-                votingMachineProposalInfo.daoBounty
-              );
-              networkCache.proposals[proposal.id].confidenceThreshold =
-                votingMachineProposalInfo.confidenceThreshold;
-              networkCache.proposals[
-                proposal.id
-              ].secondsFromTimeOutTillExecuteBoosted =
-                votingMachineProposalInfo.secondsFromTimeOutTillExecuteBoosted;
-              networkCache.proposals[proposal.id].boostedPhaseTime = bnum(
-                proposalTimes[1]
-              );
-              networkCache.proposals[proposal.id].preBoostedPhaseTime = bnum(
-                proposalTimes[2]
-              );
-              networkCache.proposals[proposal.id].daoRedeemItsWinnings =
-                votingMachineProposalInfo.daoRedeemItsWinnings;
-              networkCache.proposals[proposal.id].shouldBoost =
-                proposalShouldBoost;
-              networkCache.proposals[proposal.id].positiveVotes =
-                bnum(positiveVotes);
-              networkCache.proposals[proposal.id].negativeVotes =
-                bnum(negativeVotes);
-              networkCache.proposals[proposal.id].positiveStakes = bnum(
-                proposalStatusWithVotes[2]
-              );
-              networkCache.proposals[proposal.id].negativeStakes = bnum(
-                proposalStatusWithVotes[3]
-              );
-
-              retry = false;
-            } catch (e) {
-              console.error(
-                'Error on updating proposal (trying again)',
-                proposal
-              );
-              console.error(e);
-              retry = true;
-            }
-          }
+    await batchPromisesOntarget(
+      Object.keys(networkCache.proposals)
+        .filter(proposalId => {
+          return (
+            // Only update proposals for registered schemes
+            networkCache.schemes[networkCache.proposals[proposalId].scheme]
+              .registered &&
+            // This condition check that the proposal already existed in the current block range.
+            // If not, it means that the proposal was created in the current block range when processing the scheme,
+            // So there is no need to process it again.
+            networkCache.proposals[proposalId].creationEvent.blockNumber <
+              fromBlock &&
+            // This condition check that the proposal is active
+            (networkCache.proposals[proposalId].stateInVotingMachine >
+              VotingMachineProposalState.Executed ||
+              networkCache.proposals[proposalId].stateInScheme ===
+                WalletSchemeProposalState.Submitted)
+          );

```

Comment:
Just thinking of making the cacheService more readable and debuggable, I suggest:
```suggestion
    const proposal = networkCache.proposals[proposalId];
    const scheme = networkCache.schemes[proposal?.scheme];
    return (
      // Only update proposals for registered schemes
      scheme?.registered &&
      // This condition check that the proposal already existed in the current block range.
      // If not, it means that the proposal was created in the current block range when processing the scheme,
      // So there is no need to process it again.
      proposal?.creationEvent.blockNumber < fromBlock &&
      // This condition check that the proposal is active
      (proposal?.stateInVotingMachine > VotingMachineProposalState.Executed ||
        proposal?.stateInScheme === WalletSchemeProposalState.Submitted)
    );
```

Reply:
https://github.com/DXgovernance/dxvote/pull/795/commits/ecf96b5ea0948d9d1ea208dc8148941cf013f97b